### PR TITLE
Postgres and MySQL Schemas

### PR DESCRIPTION
**Pull request by dadrian**
Target schemas for Postgres and MySQL. One outstanding question is what to do about String() vs CensysString() when we're importing from the zgrab2 schema directly. The problem is: _Are there `String` fields in ZGrab2 schemas that should be `CensysString` in the ZTag schema, that are directly imported from ZGrab?_

@zakird, do you have thoughts on how to handle this? Is there a flag we could set to change String to look like CensysString()? Should we write some sort of recursive transformation function and do something like `zgrab2.postgres.some_subrecord | convert_string_to_censys_string`?

### CODE REVIEW

**Comment by dadrian** [comment]
Or maybe I'm totally off base here?

**Comment by dadrian** [comment]
🚨This merges into `feature/zgrab2_transforms`, not `master` 🚨

File: ztag/schema.py [hunk 1]
```diff
@@ -510,15 +510,47 @@ class CensysString(WhitespaceAnalyzedString):
...
+    "status_flags": zgrab2.mysql.mysql_server_status_flags,
+    "error_code": zgrab2.mysql.mysql_scan_response["result"]["error_code"],
+    "error_message": zgrab2.mysql.mysql_scan_response["result"]["error_message"],
+    "tls": zcrypto.TLSHandshake(doc="TODO"),
+})
+
+
+# TODO: Apply some transform to replace all String() with CensysString()?
+ztag_postgres = SubRecord({
+    "supported_versions": String(),  # TODO: CensysString?
```
Comment (zakird) [inline ztag/schema.py]: Can you provide an example of what goes into this? I'm a bit surprised to see it as a `String()` given version**s**.

File: ztag/schema.py [hunk 2]
```diff
@@ -510,15 +510,47 @@ class CensysString(WhitespaceAnalyzedString):
...
+    "protocol_error": zgrab2.postgres.postgres_error,
+    "startup_error": zgrab2.postgres.postgres_error,
+    "is_ssl": Boolean(),
+    "server_parameters": zgrab2.postgres.postgres_scan_response["server_parameters"],
+    "authentication_mode": SubRecord({
```
Comment (zakird) [inline ztag/schema.py]: Do we actually need a subrecord here?

File: ztag/schema.py (hunk 1 above)
Comment (dadrian) [reply ztag/schema.py]: @justinbastress ?

Comment (justinbastress) [reply ztag/schema.py]: This comes from an error when connecting with a bogus client version of 0.0, e.g. `"FATAL:  unsupported frontend protocol 0.0: server supports 1.0 to 3.0"`.

File: ztag/schema.py [hunk 3]
```diff
@@ -1183,6 +1213,12 @@ class LintBool(String):
                     "banner": ztag_mssql,
                 }, category="1433/MSSQL"),
             }),
+            Port(3306): SubRecord({
+                "mysql": ztag_mysql,
```
Comment (justinbastress) [inline ztag/schema.py]: Should this be `"mysql"::SubRecord({"banner":ztag_mysql})"` (and similarly for postgres), or are these different from the others?

**Pull request merged**